### PR TITLE
Make XRT compile on Ubuntu 22.04

### PR DESCRIPTION
**Pull request by Ralender**
- Fix cmake for finding python libs
  findPythonLibs is deprecated since cmake 3.12
  replace it with findPython3 for newer versions
- silence OpenSSL 3 deprecation warnings
  MD5 used in container.cpp
  and SHA256 in azure.cpp

I didn't analyze if MD5 was used for security purposes. If it is the case it should be replaced by SHA2 or better.

### CODE REVIEW

**Review by stsoe — APPROVED** [review]
Thank you for making these changes.

**Comment by stsoe** [comment]
ok to test

**Comment by Ralender** [comment]
I noticed that dbg.sh doesn't work on Ubuntu 22.04.

File: src/runtime_src/tools/xclbinutil/CMakeLists.txt [hunk 1]
```diff
@@ -3,6 +3,10 @@
 #
 # Signing xclbin images are currently only support on Linux
 
+if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.19)
+  cmake_policy(SET CMP0110 NEW)
```
Comment (keryell) [inline src/runtime_src/tools/xclbinutil/CMakeLists.txt:7]: Explain the magic with some comments.

File: src/python/pybind11/CMakeLists.txt [hunk 2]
```diff
@@ -1,23 +1,36 @@
 # SPDX-License-Identifier: Apache-2.0
 # Copyright (C) 2020-2021 Xilinx, Inc. All rights reserved.
 #
+if (${CMAKE_VERSION} VERSION_LESS "3.12")
```
Comment (keryell) [inline src/python/pybind11/CMakeLists.txt:4]: Explain the magic with some comments.

**Review by keryell — CHANGES_REQUESTED** [review]
That sounds good but you should take some rationale from your PR description into some comments in the code.

File: src/python/pybind11/CMakeLists.txt [hunk 3]
```diff
@@ -2,22 +2,37 @@
...
 #
 
-find_package(PythonLibs 3.4.0 REQUIRED)
-if (PythonLibs_FOUND)
-  message("-- Python libs version: ${PYTHONLIBS_VERSION_STRING}")
-  message("-- PYTHON_INCLUDE_PATH ${PYTHON_INCLUDE_PATH}")
-  string(REPLACE "." ";" PYTHONLIBS_VERSION_LIST ${PYTHONLIBS_VERSION_STRING})
-  list(GET PYTHONLIBS_VERSION_LIST 0 PYTHONLIBS_VERSION_MAJOR)
-  list(GET PYTHONLIBS_VERSION_LIST 1 PYTHONLIBS_VERSION_MINOR)
+# find_package(PythonLibs is depracted as of cmake 3.12.
```
Comment (keryell) [inline src/python/pybind11/CMakeLists.txt:5]: Using a spell-checker is even better :-)

**Review by keryell — COMMENTED** [review]
Thanks

**Pull request merged**